diff.py

```diff
@@ -26,30 +26,34 @@ def fail(msg):
 start_argument = parser.add_argument("start", help="Function name or address to start diffing from.")
 if argcomplete:
     def complete_symbol(**kwargs):
-        prefix = kwargs['prefix']
-        parsed_args = kwargs['parsed_args']
-        completes = []
-        config = dict()
+        prefix = kwargs["prefix"]
+        parsed_args = kwargs["parsed_args"]
+        config = {}
         diff_settings.apply(config, parsed_args)
-        mapfile = config.get('mapfile')
+        mapfile = config.get("mapfile")
         if not mapfile:
             return []
+        completes = []
         with open(mapfile) as f:
-            for line in f:
-                pos = line.find(prefix)
-                # if found the prefix at start of a word
-                # (start of line or if there's a space before)
-                if pos >= 0 and (pos == 0 or line[pos-1] == ' '):
-                    symbolEndPos = line.find(' ', pos)
-                    if symbolEndPos == -1:
-                        # -1 strips the line return '\n' (assuming Unix)
-                        symbol = line[pos:-1]
-                    else:
-                        symbol = line[pos:symbolEndPos]
-                    completes.append(symbol)
+            data = f.read()
+            if data.startswith(prefix):
+                endPos = data.find(" ")
+                completes.append(data if endPos == -1 else data[:endPos])
+            search = f" {prefix}"
```

Comment:
I didn't realize but this assumes a symbol always has a space before, can't be "\nSYMBOL"
I guess that's okay for linker maps since there are ton of spaces everywhere, such a big assumption does make the `data.startswith` case look foolish

Reply:
Yeah. Assuming a space before each symbol definitely seems fine to me. See also my comment here: https://github.com/simonlindholm/asm-differ/pull/31#discussion_r481094444